storage-local/src/main/java/io/kestra/storage/local/LocalStorage.java

```diff
@@ -176,9 +173,14 @@ public List<URI> deleteByPrefix(String tenantId, URI storagePrefix) throws IOExc
             return walk.sorted(Comparator.reverseOrder())
                 .map(r -> Pair.of(r.toFile(), r.toFile().isFile()))
                 .peek(r -> r.getLeft().delete())
-                .filter(Pair::getRight)
-                .map(r -> URI.create("kestra://" + r.getLeft().toURI().getPath().substring(config.getBasePath().toAbsolutePath().toString().length())))
-                .collect(Collectors.toList());
+                .map(r -> {
+                    String striped = r.getLeft().toURI().getPath().substring(config.getBasePath().toAbsolutePath().toString().length());
+                    if (tenantId != null) {
+                        striped = striped.substring((tenantId + "/").length());
+                    }
```

Comment:
This is the opposite of the getPath method right?
Maybe extract it to an extractPath private method for better clarity.

Also it can be done in one substring by calculating the path prefix before (with a ternary operator), it would be more readable